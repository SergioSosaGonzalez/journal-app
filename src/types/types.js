export const types = {
  login: '[Auth] login',
  logout: '[Auth] logout',

  uiSetError: '[UI] set error',
  uiRemoveError: '[UI] remove error',

  uiStartLoading: '[UI] Start loading',
  uiFinishLoading: '[UI] Finish loading',

  notesAddNew: '[Notes] New Note',
  notesActive: '[Notes] Set Active Note',
  notes: '[Notes] Load Note',
  notesUpdated: '[Notes] Update Note',
  notesLoad: '[Notes] Load Note',
  notesFileUrl: '[Notes] Updated image url',
  notesDeleted: '[Notes] Delete note',
  notesLogoutCleaning: '[Notes] Logout cleaning',
};
