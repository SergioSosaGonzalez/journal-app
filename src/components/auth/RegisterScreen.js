import React from 'react';
import { Link } from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import validator from 'validator';
import { useDispatch, useSelector } from 'react-redux';
import { removeError, setError } from '../../actions/error';
import { startRegisterWithEmailPasswordName } from '../../actions/auth';

const RegisterScreen = () => {
  const dispatch = useDispatch();
  const { msgError } = useSelector((state) => state.ui);
  let [values, handleInputChange] = useForm({
    name: '',
    email: '',
    password: '',
    confirm: '',
  });
  const { name, email, password, confirm } = values;

  const handleRegister = (e) => {
    e.preventDefault();
    if (isFormValid()) {
      dispatch(startRegisterWithEmailPasswordName(email, password, name));
    } else console.log('no registrado');
  };

  const isFormValid = () => {
    if (name.trim().length === 0) {
      dispatch(setError('Falta campo nombre'));
      return false;
    } else if (email.trim().length === 0 || !validator.isEmail(email)) {
      dispatch(setError('Email no valido'));
      return false;
    } else if (password !== confirm || password.length < 5) {
      dispatch(setError('La contraseÃ±a debe ser menor a 5 caracteres'));
      return false;
    }
    dispatch(removeError());
    return true;
  };
  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form
        onSubmit={handleRegister}
        className="animate__animated animate__fadeIn animate__faster">
        {msgError && <div className="auth__alert-error">{msgError}</div>}
        <input
          type="tex"
          placeholder="Name"
          className="auth__input"
          name="name"
          autoComplete="off"
          value={name}
          onChange={handleInputChange}
        />
        <input
          type="email"
          placeholder="email"
          className="auth__input"
          name="email"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />
        <input
          type="password"
          placeholder="Password"
          className="auth__input"
          name="password"
          value={password}
          onChange={handleInputChange}
        />
        <input
          type="password"
          placeholder="Confirm Password"
          className="auth__input"
          name="confirm"
          value={confirm}
          onChange={handleInputChange}
        />
        <button type="submit" className="btn btn-primary btn-block mb-5">
          Register
        </button>
        <Link to="/auth/login" className="link">
          Are you already register?
        </Link>
      </form>
    </>
  );
};
export default RegisterScreen;
