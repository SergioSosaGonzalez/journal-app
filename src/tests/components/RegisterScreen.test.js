import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { types } from '../../types/types';
import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';
import RegisterScreen from '../../components/auth/RegisterScreen';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
  auth: {},
  ui: {
    loading: false,
    msgError: null,
  },
};

let store = mockStore(initState);

describe('Pruebas para RegisterScreen', () => {
  beforeEach(() => {
    store = mockStore(initState);
  });
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <RegisterScreen />
      </MemoryRouter>
    </Provider>
  );
  test('Debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de hacer el dispatch de la accion respectiva', () => {
    const emailField = wrapper.find('input[name="email"]');
    emailField.simulate('change', { target: { value: '', name: 'email' } });
    wrapper.find('form').prop('onSubmit', { preventDefault() {} });
    const actions = store.getActions();
  });

  test('Debe de mostrar un mensaje de error', () => {
    const initState = {
      auth: {},
      ui: {
        loading: false,
        msgError: 'Error de algo',
      },
    };
    let store = mockStore(initState);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <RegisterScreen />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper.find('.auth__alert-error').exists()).toBe(true);
    expect(wrapper.find('.auth__alert-error').text().trim()).toBe(
      initState.ui.msgError
    );
  });

  test('Debe de mostrar la caja con el error', () => {
    const initState = {
      auth: {},
      ui: {
        loading: false,
        msgError: null,
      },
    };
    
    let store = mockStore(initState);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <RegisterScreen />
        </MemoryRouter>
      </Provider>
    );
  })
  
});
