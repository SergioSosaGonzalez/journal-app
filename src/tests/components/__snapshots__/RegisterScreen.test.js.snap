// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para RegisterScreen Debe de mostrarse correctamente 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Register
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="tex"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="email"
      value=""
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="auth__input"
      name="confirm"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
      value=""
    />
    <button
      className="btn btn-primary btn-block mb-5"
      type="submit"
    >
      Register
    </button>
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Are you already register?
    </a>
  </form>,
]
`;
